Examples/Docs/Sources/AsyncExamples/main.swift

```diff
@@ -225,3 +225,54 @@ private func transactions() throws {
     }.wait()
     // End Transactions Retry Example 3
 }
+
+private func versionedAPI() throws {
+    let uri = "mongodb://localhost:27017"
+    let myEventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+    defer { try? myEventLoopGroup.syncShutdownGracefully() }
+
+    do {
+        // Start Versioned API Example 1
+
+        // Declare API version "1" for the client
+        var clientOpts = MongoClientOptions()
+        clientOpts.serverAPI = MongoServerAPI(version: .v1)
+        let client = try MongoClient(uri, using: myEventLoopGroup, options: clientOpts)
+        // End Versioned API Example 1
+
+        try client.syncClose()
+    }
+
+    do {
+        // Start Versioned API Example 2
+
+        // Use the `strict` option
+        var opts = MongoClientOptions()
+        opts.serverAPI = MongoServerAPI(version: .v1, strict: true)
+        let client = try MongoClient(uri, using: myEventLoopGroup, options: opts)
+
+        var findOpts = FindOptions()
+        findOpts.cursorType = .tailable
+        // Fails with an error because `tailable` is not part of version 1
+        do {
+            let cursor = try client.db("db").collection("coll").find(options: findOpts).wait()
```

Comment:
would it be better to write this using a `.whenFailure` callback to more closely simulate async error handling?

Reply:
updated. I added a `try? cursor.wait()` down after the example to ensure the future actually finishes before we close the client. I debated putting it in the example, but it didn't seem particularly relevant